libraries/slave_windows.rb

```diff
@@ -81,9 +80,24 @@ def action_create
     end
 
     protected
-
+    
     # Embedded Resources
 
+    # Creates a `directory` resource that represents the directory
+    # specified the `remote_fs` attribute. The caller will need to call
+    # `run_action` on the resource.
+    #
+    # @return [Chef::Resource::Directory]
+    #
+    def remote_fs_dir_resource
+      return @remote_fs_dir_resource if @remote_fs_dir_resource
```

Comment:
@ksubrama  I think that might be a better idea.  Let me know if you agree with the reasoning above, or if we should just make this change alot smaller and override the perms method within the windows class and call it a day.

As stated above, thought the code was a bit more straight forward when separating the constructor logic for unix/windows.